### DISCUSSION TITLE
First arg of `clojure.core/get` should be associative

### DISCUSSION THREAD

**Discussion by vemv** [borkdude/speculative#315] [Ideas]
Currently it's defined as a `::ss/any`. If it was defined as an 'associative', it would avoid the pitfaill described in this discussion:

https://ask.clojure.org/index.php/1917/get-should-throw-exception-on-non-associative-argument

---

(n.b. I'm just reporting this as an idea for anyone that might want to work on this project in a future. I'm not currently a user)

**Comment by borkdude** [comment]
We spec'ed it as `any?` since `get` takes any value, including `ILookup` (rather than `IAssociative`) and typically looks something up, but when it doesn't implement that interface, returns `nil`. So anything is a valid input. Spec'ing it otherwise would be overly restrictive.

**Reply by vemv** [reply]
But if you follow that discussion, you'll note that multiple contributors noted this is as undesirable behavior. Alex Miller in particular noted that a spec could guard against this design idiosyncrasy.

It seems to me that Speculative has the most usefulness when it avoids bugs, rather than perpetuate them.

e.g. what's the use of `(get "s" :k)`?

Thanks - V

**Reply by borkdude** [reply]
`get` on String is explicitly supported here: https://github.com/clojure/clojure/blob/5ffe3833508495ca7c635d47ad7a1c8b820eab76/src/jvm/clojure/lang/RT.java#L810
but yeah, the keyword value doesn't make sense with a string... :)

**Reply by vemv** [reply]
TIL!

`(get (atom nil) :k)` then, for leaving a valid example for posterity 😇
